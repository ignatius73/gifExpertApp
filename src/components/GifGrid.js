import React, { useEffect } from 'react';

export const GifGrid = ({category}) => {

    useEffect( () => {
        getGiphes();
    }, []);

    const getGiphes = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q={category}&limit=10&api_key=B4VMClSog6hZwFQ5i6KsnU7CgUxUWeDJ'
        const resp = await fetch(url);
        const { data } = await resp.json();
        
        const gifs = data.map( img => {
            return{
                id: img.id,
                gif: img.images?.downsized_large.url,
                title: img.title   
            }
        });
            
        console.log(gifs);
        
    }

    getGiphes();
    return (
                <>
                    <h3>{category}</h3>
                </>
            )
}
