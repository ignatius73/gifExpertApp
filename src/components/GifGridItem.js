import React from 'react'

export const GifGridItem = (props) => {

    console.log(props);

    return (
        <div className='card animate__animated animate__bounce animate__jackInTheBox'>
          {  <img src={props.gif} alt={props.title} /> }
        </div>
    )
}
