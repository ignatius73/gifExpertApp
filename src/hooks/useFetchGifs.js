import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = ({ category }) => {
    const [state, setstate] = useState({
        data:[],
        loading: true});

    useEffect( () => {
        getGifs({category}).then( (gifs) =>{
            setTimeout( () =>{
                setstate({
                    data: gifs,
                    loading: false
                    });

            },3000);
            
        });
        
        },[category]);
    
    return state;
    
}

