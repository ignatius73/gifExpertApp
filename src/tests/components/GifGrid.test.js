import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');




describe('Unitary tests over GifGrid.js', () => {

    const categ = 'One Punch';
    //let wrapper = shallow( <GifGrid category={ categ } /> );

   /* beforeEach( ()=>{

    })*/

    test('should match with the snapshot', () => {
        //Controla la llegada por default del hookw

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={ categ } /> );
        expect( wrapper ).toMatchSnapshot();
        
       
    })

    test('should be show items when images were loaded', () => {
        //Creo un objeto de las características que debería recibir del Fetch
        const gifs = [{
            id: 'ABC',
            gif:'http://cualqierCosa.com/imagen.jpg',
            title: 'Cualquier cosa'
        }];

        //Llamo el returnValue del Hook
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ categ } />);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
    
    
});
