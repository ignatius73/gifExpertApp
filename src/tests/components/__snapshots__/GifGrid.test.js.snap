// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitary tests over GifGrid.js should match with the snapshot 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
   
  <hr />
   
  <p>
     Loading... 
  </p>
   
  <div
    className="card-grid"
  />
    
</Fragment>
`;
